crates/light-client-prover/src/da_block_handler.rs

```diff
@@ -149,10 +149,27 @@ where
 
         // Do any kind of ordering etc. on batch proofs here
         // If you do so, don't forget to do the same inside zk
+        let batch_proof_method_id = self
+            ._batch_proof_code_commitments_by_spec
+            .get(&SpecId::Genesis)
+            .expect("Batch proof code commitment not found");
+
+        let mut assumptions = vec![];
+        let mut journals = vec![];
+        for batch_proof in batch_proofs {
+            if let DaDataLightClient::Complete(proof) = batch_proof {
+                // Handle err
+                let output = Vm::verify(proof.as_slice(), batch_proof_method_id).unwrap();
```

Comment:
I believe we don't necessarily need to verify here. We can just use extract_output or some similar method, no?

Reply:
Should we not verify the proofs at some point tho? Check out the new version

Reply:
but we verify already in circuit. we just need the output.

one benefit would be to not start proving if we see beforehand that its broken. I am not sure what to do with that do.

Reply:
assertion here is fine imo, but expect instead of unwrap could be slightly better.

Reply:
> one benefit would be to not start proving if we see beforehand that its broken. I am not sure what to do with that neither.

we should verify for all sorts of reasons, trust me